src/miral/wayland_extensions.cpp

```diff
@@ -40,6 +42,52 @@ auto vec2set(std::vector<std::string> vec) -> std::set<std::string>
 {
     return std::set<std::string>{vec.begin(), vec.end()};
 }
+
+struct StaticExtensionTracker
```

Comment:
Can we get some documentation on this as to what it is and why it's needed? `StaticExtensionTracker` does not make it's purpose immediately obvious.